### ISSUE TITLE
Support for "multinom" in the ate function

### ISSUE THREAD

**Issue by RobinDenz1** [tagteam/riskRegression#21]
Dear Prof. Brice Ozenne,

I was wondering if the `ate` function supports categorical treatment variables when method is set to "IPTW" or "AIPTW". It works perfectly fine when using the G-Formula approach. It is not stated in the documentation that it does, but I got the following warning message when I accidentally input a `glm` model with a categorical treatment:

`Cannot use a logistic regression in argument "treatment" when there are more than two treatment categories 
Consider subsetting the dataset to have only two treatment categories or using a multinomial regression (nnet::multinom).`

Here is some code using an example simulated dataset (uploaded with this post) to reproduce the warning message:
```
library(riskRegression)
library(nnet)

# the data
sim_dat <- read.csv("Simulated_Data.csv", sep="\t")

# reproducing the warning message
sim_dat$group2 <- sample(0:1, size=nrow(sim_dat), replace=T)

wrong_glm <- glm(group2 ~ x2 + x4 + x5, data=sim_dat)

curve <- riskRegression::ate(event=outcome_model,
                             treatment=wrong_glm,
                             censor=censoring_model,
                             data=sim_dat,
                             times=1:1000,
                             se=F,
                             estimator="AIPTW,AIPCW")
```
So naturally I wanted to use a multinomial logistic regression model as described in the warning message:
```
multin <- nnet::multinom(group ~ x2 + x4 + x5, data=sim_dat)

outcome_model <- survival::coxph(Surv(time, cause) ~ ., data=sim_dat, x=T)

censoring_model <- survival::coxph(Surv(time, cause==0) ~ 1, data=data, x=T)

curve <- riskRegression::ate(event=outcome_model,
                             treatment=multin,
                             censor=censoring_model,
                             data=sim_dat,
                             times=1:1000,
                             se=F,
                             estimator="AIPTW,AIPCW")
```
This however only produces the following error message: "Error in ate_initArgs(object.event = event, object.treatment = treatment,  : Objekt 'treatment' not found". I took a look at the source code of the `ate_initArgs` function and saw that it doesn't support objects of class `multinom`.

My question is: Is the support of `multinom` not implemented yet and the warning message is outdated or is this a bug?

[Simulated_Data.zip](https://github.com/tagteam/riskRegression/files/5924908/Simulated_Data.zip)

**Comment by bozenne** [comment]
Dear Robin Denz,

The ate function has been tested for binary treatment variable. For categorical treatment variables I have implemented something that should work but unfortunately not had time to perform any simulation study.
This being said, there was indeed a bug when using IPTW. I have fixed this in the new version on github (2021.02.04, branch master). That should solve the issue.

You can try the following code:
`
library(riskRegression)
library(nnet)
library(survival)

# the data
sim_dat <- read.csv("Simulated_Data.csv", sep="\t")
sim_dat$group <- as.factor(sim_dat$group)
sim_dat$group01 <- as.factor(sim_dat$group %in% c("0","1"))
sim_dat$group02 <- as.factor(sim_dat$group %in% c("0","2"))

## time horizon
tau <- c(10,50,100,500,1000)

## comparing 0 vs. 1
multin01 <- nnet::multinom(group01 ~ x2 + x4 + x5, data=sim_dat)
glm01 <- glm(group01 ~ x2 + x4 + x5, family = binomial(link = "logit"), data=sim_dat)
outcome_model01 <- survival::coxph(Surv(time, cause) ~ x1+x2+x3+x4+x5+group01, data=sim_dat, x=T)
censoring_model01 <- survival::coxph(Surv(time, cause==0) ~ 1, data=sim_dat, x=T)

ate.glm01 <- ate(event=outcome_model01,
                 treatment=glm01,
                 censor=censoring_model01,
                 data=sim_dat,
                 times=tau,
                 estimator="AIPTW,AIPCW")
confint(ate.glm01)$meanRisk[,c("estimate","se")]

ate.multinom01 <- ate(event=outcome_model01,
                      treatment=multin01,
                      censor=censoring_model01,
                      data=sim_dat,
                      times=tau,
                      estimator="AIPTW,AIPCW")

## should be the same
range(confint(ate.glm01)$meanRisk[,c("estimate","se")] - confint(ate.multinom01)$meanRisk[,c("estimate","se")])

## comparing 0 vs. 2
multin02 <- nnet::multinom(group02 ~ x2 + x4 + x5, data=sim_dat)
glm02 <- glm(group02 ~ x2 + x4 + x5, family = binomial(link = "logit"), data=sim_dat)
outcome_model02 <- survival::coxph(Surv(time, cause) ~ x1+x2+x3+x4+x5+group02, data=sim_dat, x=T)
censoring_model02 <- survival::coxph(Surv(time, cause==0) ~ 1, data=sim_dat, x=T)

ate.glm02 <- ate(event=outcome_model02,
                 treatment=glm02,
                 censor=censoring_model02,
                 data=sim_dat,
                 times=tau,
                 estimator="AIPTW,AIPCW")
confint(ate.glm02)$meanRisk[,c("estimate","se")]

ate.multinom02 <- ate(event=outcome_model02,
                      treatment=multin02,
                      censor=censoring_model02,
                      data=sim_dat,
                      times=tau,
                      estimator="AIPTW,AIPCW")

## should be the same
range(confint(ate.glm02)$meanRisk[,c("estimate","se")] - confint(ate.multinom02)$meanRisk[,c("estimate","se")])

## comparing all treatments
multin <- nnet::multinom(group ~ x2 + x4 + x5, data=sim_dat)
outcome_model <- survival::coxph(Surv(time, cause) ~ x1+x2+x3+x4+x5+group, data=sim_dat, x=T)
censoring_model <- survival::coxph(Surv(time, cause==0) ~ 1, data=sim_dat, x=T)

ate.multinom <- ate(event=outcome_model,
                    treatment=multin,
                    censor=censoring_model,
                    data=sim_dat,
                    times=tau,
                    estimator="AIPTW,AIPCW")

## not expected to match in finite samples
## but should be the same in large samples if the multinom model is valid
confint(ate.glm01)$meanRisk[,c("estimate","se")] - confint(ate.multinom)$meanRisk[confint(ate.multinom)$meanRisk$treatment %in% c("0","1"),c("estimate","se")]
confint(ate.glm02)$meanRisk[,c("estimate","se")] - confint(ate.multinom)$meanRisk[confint(ate.multinom)$meanRisk$treatment %in% c("0","2"),c("estimate","se")]
`

**Comment by RobinDenz1** [comment]
Thank you a lot for the fast and detailed answer! This resolves the issue for me.
